File: 2233-maximum-product-after-k-increments/2233-maximum-product-after-k-increments.cpp
class Solution {
public:
    int mod = 1000000007;
    int maximumProduct(vector<int>& nums, int k) {
        int n = nums.size();
        priority_queue<int, vector<int>, greater<int>> pq;
        for(int i : nums) pq.push(i);
        
        while(k--) {
            int x = pq.top()+1;
            // cout<<"dbg : "<<x<<endl;
            pq.pop();
            pq.push(x);
        }
        long long ans = 1;
        while(!pq.empty()) {
            ans = (ans*1LL*pq.top())%mod;
            pq.pop();
        }
        return ans%mod;
    }
};

File: 2233-maximum-product-after-k-increments/2233-maximum-product-after-k-increments.java
class Solution {
    public int maximumProduct(int[] nums, int k) {
        int n = nums.length;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for(int i : nums) pq.add(i);
        
        while(k != 0) {
            int x = pq.poll();
            pq.add(x+1);
            k--;
        }
        long ans = 1;
        int mod = 1000000007;
        while(!pq.isEmpty()) {
            ans = (ans*pq.peek())%mod;
            pq.poll();
        }
        return (int)ans;
    }
}